integration-libs/opf/checkout/components/opf-checkout-payment-wrapper/opf-checkout-payment-wrapper.service.ts

```diff
@@ -0,0 +1,270 @@
+    this.opfService.updateOpfMetadataState({ paymentSessionId });
+  }
+
+  reloadPaymentMode(): void {
+    if (this.lastPaymentOptionId) {
+      this.initiatePayment(this.lastPaymentOptionId).subscribe();
+    }
+  }
+
+  renderPaymentGateway(config: PaymentSessionData) {
+    if (config?.destination) {
+      this.renderPaymentMethodEvent$.next({
+        isLoading: false,
+        isError: false,
+        renderType: OpfPaymentMethodType.DESTINATION,
+        data: config?.destination.url,
+        destination: config?.destination,
+      });
+    }
+
+    if (config?.dynamicScript) {
+      const html = config?.dynamicScript?.html;
+
+      this.opfResourceLoaderService
+        .loadProviderResources(
+          config.dynamicScript.jsUrls,
+          config.dynamicScript.cssUrls
+        )
+        .then(() => {
+          this.renderPaymentMethodEvent$.next({
+            isLoading: false,
+            isError: false,
+            renderType: OpfPaymentMethodType.DYNAMIC_SCRIPT,
+            data: html,
+          });
+
+          if (html) {
+            this.executeScriptFromHtml(html);
+          }
+        });
+    }
+  }
+
+  protected handlePaymentInitiationError(
+    err: HttpErrorModel
+  ): Observable<Error> {
+    if (isAuthorizationError(err)) {
+      return throwError(() => err);
+    }
+
+    return Number(err.status) === HttpResponseStatus.CONFLICT
+      ? this.handlePaymentAlreadyDoneError()
+      : this.handleGeneralPaymentError();
+  }
+
+  protected handlePaymentAlreadyDoneError(): Observable<Error> {
+    return this.opfOrderFacade.placeOpfOrder(true).pipe(
+      catchError(() => {
+        this.onPlaceOrderError();
+
+        // If place order will fail after two attempts, we wan't to stop stream and show error message
+        return of();
+      }),
+      switchMap(() => {
+        this.onPlaceOrderSuccess();
+
+        return throwError('Payment already done');
+      })
+    );
+  }
+
+  protected onPlaceOrderSuccess(): void {
+    this.routingService.go({ cxRoute: 'orderConfirmation' });
+  }
+
+  protected onPlaceOrderError(): void {
+    this.renderPaymentMethodEvent$.next({
+      ...this.renderPaymentMethodEvent$.value,
+      isError: true,
+    });
+
+    this.showErrorMessage('opf.checkout.errors.unknown');
+    this.routingService.go({ cxRoute: OPF_PAYMENT_AND_REVIEW_SEMANTIC_ROUTE });
+  }
+
+  protected handleGeneralPaymentError(): Observable<Error> {
+    this.renderPaymentMethodEvent$.next({
+      ...this.renderPaymentMethodEvent$.value,
+      isError: true,
+    });
+
+    this.showErrorMessage('opf.payment.errors.proceedPayment');
+
+    return throwError('Payment failed');
+  }
+
+  protected showErrorMessage(errorMessage: string): void {
+    this.globalMessageService.add(
+      {
+        key: errorMessage,
+      },
+      GlobalMessageType.MSG_TYPE_ERROR
+    );
+  }
+
+  protected setPaymentInitiationConfig(
+    otpKey: string,
+    paymentOptionId: number
+  ) {
+    return {
```

Comment:
[MEDIUM]

`set...` vs `get...`? This method returns this config, but not make side effect setting

Reply:
[OUTDATED] Done.